apiserver/facades/agent/provisioner/provisioner.go

```diff
@@ -37,6 +39,23 @@ import (
 	"github.com/juju/juju/storage/poolmanager"
 )
 
+// ControllerConfigGetter defines the methods required to get the controller
+type ControllerConfigGetter interface {
+	ControllerConfig(stdcontext.Context) (controller.Config, error)
+}
```

Comment:
I'm thinking these interfaces should be `ControllerConfigService`. So let's try and and standardise on this.